### PR TITLE
print profile summary and test summary

### LINKED ISSUE #852: InSpec output for summary needs to count controls
Currently InSpec counts "examples", i.e. so-called RSpec examples.
We want to drop that as a visible concept and as a term in our vocabulary book.

InSpec counts controls and tests. Controls may have a number of tests. Tests may exist without controls (albeit: internally we map them to anonymous controls anyway).

InSpec's output reflects RSpec examples (i.e. tests): Count of failed, passed, and skipped tests.
## Suggestion

Number of controls (total), number of passed controls, failed controls, number of skipped controls.
(Optional maybe: number of tests, passed tests, failed tests, skipped tests. (May not be necessary!!))
Controls that exist but don't have any tests, is counted as a control in the list of total controls, but not as a passed or failed or skipped control.

### PR DESCRIPTION
**Pull request by vjeffrey**
i put a wip on it only because I'm working on getting the two shell functional tests that are failing to a passing state; i'm a little confused about why they're failing, since running the same thing those tests are running manually gives me the correct output.
screen shot 2016-09-15 at 4 50 47 pm
screen shot 2016-09-15 at 4 50 55 pm
screen shot 2016-09-15 at 4 51 02 pm

fixes https://github.com/chef/inspec/issues/852

### CODE REVIEW

**Comment by vjeffrey** [comment]
this is what the two failing shell tests are running; it gives me the correct output here but comes back with an incorrect summary count in the functional tests  :/  
screen shot 2016-09-15 at 4 55 50 pm

**Comment by stevendanna** [comment]
@vjeffrey I think the remaining failures are related to `inspec shell -c` which takes a whole different code path: https://github.com/chef/inspec/blob/master/lib/inspec/cli.rb#L214-L224

**Comment by vjeffrey** [comment]
argh!!!!!! ok, i'll go try to fix that. argggg

**Comment by vjeffrey** [comment]
mo-flippin-o i'm down to one failing test. almost there.

**Comment by vjeffrey** [comment]
screen shot 2016-09-15 at 7 09 48 pm
AHHHHHHHHHHHHHHHH
i'm trying to figure out why the second test isn't in the results array....

**Comment by vjeffrey** [comment]
ready for review!

File: lib/inspec/rspec_json_formatter.rb [hunk 1]
```diff
@@ -133,34 +130,72 @@ def stop(notification)
...
 
-    @profiles_info.each do |_name, profile|
-      total += profile[:controls].length
-      profile[:controls].each do |_control_name, control|
-        next unless control[:results]
-        if control[:results].any? { |r| r[:status] == 'failed' }
-          failed += 1
-        elsif control[:results].any? { |r| r[:status] == 'skipped' }
-          skipped += 1
+    @anonymous_tests.each do |control|
```
Comment (arlimus) [inline lib/inspec/rspec_json_formatter.rb]: Please count control_tests too :)

File: lib/inspec/rspec_json_formatter.rb [hunk 2]
```diff
@@ -133,34 +130,72 @@ def stop(notification)
...
+      next unless control[:results]
+      control[:results].each do |result|
+        if result[:status] == 'failed'
+          failed +=1
+        elsif result[:status] == 'skipped'
+          skipped +=1
         else
-          passed += 1
+          passed +=1
```
Comment (arlimus) [inline lib/inspec/rspec_json_formatter.rb]: please add spaces

File: lib/inspec/rspec_json_formatter.rb [hunk 3]
```diff
@@ -266,13 +302,26 @@ def close(_notification)
...
+
     s = format('Summary: %s%d successful%s, %s%d failures%s, %s%d skipped%s',
-               COLORS['passed'], passed, COLORS['reset'],
-               COLORS['failed'], res[:failure_count], COLORS['reset'],
-               COLORS['skipped'], res[:skip_count], COLORS['reset'])
-    output.puts(s)
+               COLORS['passed'], 0, COLORS['reset'],
+               COLORS['failed'], 0, COLORS['reset'],
+               COLORS['skipped'], 0, COLORS['reset'])
+    output.puts(s) if @current_control.nil? || (@anonymous_tests.empty? && controls_res['total'] == 0)
```
Comment (arlimus) [inline lib/inspec/rspec_json_formatter.rb]: Please remove this old-school summary and use the Test summary instead (adjusting functional tests)

**Comment by vjeffrey** [comment]
comments addressed @arlimus :)

**Comment by vjeffrey** [comment]
screen shot 2016-09-18 at 9 44 37 pm

**Comment by arlimus** [comment]
Awesome improvement, thank you @vjeffrey !! :smile: 
Also kudos @stevendanna for helping on this one.

**Pull request merged**